src/Lecture1.hs

```diff
@@ -94,7 +101,13 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> [Char] -> [Char]
+subString start end str =
+    let newStart = if start < 0 then 0 else start
```

Comment:
You can use a standard function `max` to write this shorter

```suggestion
    let newStart = max 0 start
```

Reply:
That's much nicer 👍